This file is part of MXE. See LICENSE.md for licensing information.

Contains ad hoc patches for cross building.

From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Marco Borrini <borrinimarco74@gmail.com>
Date: Tue, 5 Jul 2016 21:12:38 +0300
Subject: [PATCH 1/1] fix for deprecated mkdir

The POSIX mkdir(directory, mode) is deprecated or not available with the
same signature on Windows/MinGW. This patch replaces it with the Windows
specific _mkdir(directory) when compiling for Windows.

diff -ur libxslt-orig/libxslt/security.c libxslt-new/libxslt/security.c
--- libxslt-orig/libxslt/security.c
+++ libxslt-new/libxslt/security.c
@@ -327,7 +327,11 @@ xsltCheckWritePath(xsltSecurityPrefsPtr sec,
 	    }
 	    ret = xsltCheckWritePath(sec, ctxt, directory);
 	    if (ret == 1)
-		ret = mkdir(directory, 0755);
+#if defined(_WIN32) || defined(__MINGW32__)
+		ret = _mkdir(directory);
+#else
+		ret = mkdir(directory, 0755);
+#endif
 	}
 	xmlFree(directory);
 	if (ret < 0)
